pkg/kubelet/network/dns/dns.go

```diff
@@ -353,6 +353,26 @@ func mergeDNSOptions(existingDNSConfigOptions []string, dnsConfigOptions []v1.Po
 	return options
 }
 
+// appendOptions appends options to the given list, but does not add duplicates.
+// append option will overwrite the previous one either in new line or in the same line.
+func appendOptions(options []string, newOption ...string) []string {
+	var optionMap = make(map[string]string)
+	for _, option := range options {
+		optName := strings.Split(option, ":")[0]
```

Comment:
we must not assume that resolv.conf is well formatted. please test the length of option after splitting it.

Reply:
> we must not assume that resolv.conf is well formatted. please test the length of option after splitting it.

But I use the first element, there can be no ":" in options. 
`option edns0` is also valid.

Reply:
Use strings.Cut instead.